### PR TITLE
Implement Vector3::maxComponents and Vector3::minComponents

### PR DESCRIPTION
**Pull request by CortexPE**
In comparison to #5, this implementation allows for an indefinite count of vectors to be passed, instead of being limited on just two.

### CODE REVIEW

**Comment by dktapps** [comment]
Please add unit tests so that we can see this is working as intended.

File: tests/phpunit/Vector3Test.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+namespace pocketmine\math;
+
+use PHPUnit\Framework\TestCase;
+
+class Vector3Test extends TestCase {
```
Comment (dktapps) [inline tests/phpunit/Vector3Test.php]: pls remove extra space before {

Comment (CortexPE) [reply tests/phpunit/Vector3Test.php]: https://github.com/pmmp/Math/blob/1bf44397897d8e50d747e9e7a3de245c21b464c9/tests/phpunit/FacingTest.php#L26-L28

:thinking:

Comment (CortexPE) [reply tests/phpunit/Vector3Test.php]: o nvm ree

File: src/Vector3.php [hunk 2]
```diff
@@ -421,6 +429,14 @@ public static function max(Vector3 ...$positions) : Vector3{
 		return new Vector3(max($xList), max($yList), max($zList));
 	}
 
+	/**
+	 * Returns a new vector with the minimum xyz coordinates based from the passed vectors.
+	 * Behavior is similar with the standard PHP min function < https://php.net/manual/en/function.min.php >
```
Comment (SOF3) [inline src/Vector3.php]: I don't think it is in any sense similar to the PHP min function. Removing this line may lead to less confusion. PHP min function returns one of the inputs, while this doesn't.

File: src/Vector3.php [hunk 3]
```diff
@@ -421,6 +429,14 @@ public static function max(Vector3 ...$positions) : Vector3{
 		return new Vector3(max($xList), max($yList), max($zList));
 	}
 
+	/**
+	 * Returns a new vector with the minimum xyz coordinates based from the passed vectors.
```
Comment (SOF3) [inline src/Vector3.php]: It is more clear to say "Returns a new Vector3 taking the minimum of each component in the input vectors."
Another way to express this is "applies the PHP min function on each component over the input vectors".

Comment (dktapps) [reply src/Vector3.php]: Perhaps minComponents() might be a better name with this in mind.

**Comment by dktapps** [comment]
I think this is good to merge now. @SOF3 do you want to re-review?

**Pull request merged**